hazelcast/src/hazelcast/client/impl/statistics/Statistics.cpp

```diff
@@ -0,0 +1,234 @@
+
+                                    std::string prefix = nearCacheNameWithPrefix.str();
+
+                                    addStat(stats, prefix, "creationTime", nearCacheStats.getCreationTime());
+                                    addStat(stats, prefix, "evictions", nearCacheStats.getEvictions());
+                                    addStat(stats, prefix, "hits", nearCacheStats.getHits());
+                                    addStat(stats, prefix, "lastPersistenceDuration",
+                                            nearCacheStats.getLastPersistenceDuration());
+                                    addStat(stats, prefix, "lastPersistenceKeyCount",
+                                            nearCacheStats.getLastPersistenceKeyCount());
+                                    addStat(stats, prefix, "lastPersistenceTime",
+                                            nearCacheStats.getLastPersistenceTime());
+                                    addStat(stats, prefix, "lastPersistenceWrittenBytes",
+                                            nearCacheStats.getLastPersistenceWrittenBytes());
+                                    addStat(stats, prefix, "misses", nearCacheStats.getMisses());
+                                    addStat(stats, prefix, "ownedEntryCount", nearCacheStats.getOwnedEntryCount());
+                                    addStat(stats, prefix, "expirations", nearCacheStats.getExpirations());
+                                    addStat(stats, prefix, "invalidations", nearCacheStats.getInvalidations());
+                                    addStat(stats, prefix, "invalidationRequests",
+                                            nearCacheStats.getInvalidationRequests());
+                                    addStat(stats, prefix, "ownedEntryMemoryCost",
+                                            nearCacheStats.getOwnedEntryMemoryCost());
+                                    std::string persistenceFailure = nearCacheStats.getLastPersistenceFailure();
+                                    if (!persistenceFailure.empty()) {
+                                        addStat(stats, prefix, "lastPersistenceFailure", persistenceFailure);
+                                    }
+                                }
+
+                }
+
+                Statistics::PeriodicStatistics::PeriodicStatistics(Statistics &statistics) : statistics(statistics) {}
+
+                void Statistics::PeriodicStatistics::getNameWithPrefix(const std::string &name, std::ostringstream &out) {
+                    out << NEAR_CACHE_CATEGORY_PREFIX;
+
+                    std::string escapedName = boost::replace_all_copy(name, ",", "\\,");
+                    boost::replace_all(escapedName, "=", "\\=");
+                    boost::replace_all(escapedName, "\\", "\\\\");
+
+                    if (name[0] == '/') {
```

Comment:
`I don't remember the reason for removing this character at the beginning`

I think this is exactly why there should be a method with a descriptive name so that when someone else or even you look at the code can understand why it is there and it is much easier to maintain.

Reply:
Ok, changed. Put a method. `static std::string escapeSpecialCharacters(const std::string &name)`